modules/compute/vm-instance/startup_from_network_storage.tf

```diff
@@ -52,11 +52,15 @@ locals {
   )
 
   startup_from_network_storage = module.netstorage_startup_script.startup_script
+
+  service_account_email = lookup(var.service_account, "email", null)
+  bucket_viewers        = local.service_account_email != null && var.set_bucket_viewer ? ["serviceAccount:${local.service_account_email}"] : []
```

Comment:
If it is null, we can be pretty sure that they are using the default compute SA. The default compute SA may not have permission to the bucket. What are your thoughts on granting the role to the default compute SA in this case? I think it can be retrieved via a data block.

Reply:
My preference is to move toward a model similar to `pre-existing-vpc` in which we have a "default-compute-service-account" module which wraps the data source you're thinking of. Then `null` should not be an allowed value for `var.service_account_email` in the first place.

We have previously identified it as a problem that (sub)network_self_link taking on null in the vm-instance module will "default to the default network". We should identify service account null-ness similarly.

In this model, the default SA is "just another SA" and there are no code/documentation loops to understand.

Reply:
Resolving due to lower scope of PR